core/src/main/java/io/kestra/plugin/core/trigger/Schedule.java

```diff
@@ -630,5 +627,23 @@ public static class Output implements io.kestra.core.models.tasks.Output {
         @Schema(title = "The date of the previous schedule")
         @NotNull
         private ZonedDateTime previous;
+        
+        public Map<String, Object> toMap(ZoneId zoneId) {
+            Map<String, Object> map = new HashMap<>();
+            if (date != null) {
+                map.put("date", date.withZoneSameInstant(zoneId));
+            }
+            if (next != null) {
+                map.put("next", next.withZoneSameInstant(zoneId));
+            }
+            if (previous != null) {
+                map.put("previous", previous.withZoneSameInstant(zoneId));
+            }
+            return map;
+        }
```

Comment:
I would prefer that this would be done when we compute the date instead of when we serialize them.
You should not override toMap() or provide an alternative way to create the Map of output.
When you create the Output, you should pass it the correctly computed dates.